<!DOCTYPE html>
<html lang="en"><head><meta charset="UTF-8"/><meta name="viewport" content="width=device-width, initial-scale=1.0"/><title>Full API · ChemometricsTools</title><link href="https://cdnjs.cloudflare.com/ajax/libs/normalize/4.2.0/normalize.min.css" rel="stylesheet" type="text/css"/><link href="https://fonts.googleapis.com/css?family=Lato|Roboto+Mono" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/default.min.css" rel="stylesheet" type="text/css"/><script>documenterBaseURL="../.."</script><script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.2.0/require.min.js" data-main="../../assets/documenter.js"></script><script src="../../siteinfo.js"></script><script src="../../../versions.js"></script><link href="../../assets/documenter.css" rel="stylesheet" type="text/css"/></head><body><nav class="toc"><h1>ChemometricsTools</h1><select id="version-selector" onChange="window.location.href=this.value" style="visibility: hidden"></select><form class="search" id="search-form" action="../../search/"><input id="search-query" name="q" type="text" placeholder="Search docs"/></form><ul><li><a class="toctext" href="../../">Home</a></li><li><span class="toctext">Demos</span><ul><li><a class="toctext" href="../../Demos/Transforms/">Transforms</a></li><li><a class="toctext" href="../../Demos/Pipelines/">Pipelines</a></li><li><a class="toctext" href="../../Demos/ClassificationExample/">Classification</a></li><li><a class="toctext" href="../../Demos/RegressionExample/">Regression</a></li><li><a class="toctext" href="../../Demos/SIPLS/">SIPLS</a></li><li><a class="toctext" href="../../Demos/CalibXfer/">Calibration Transfer</a></li><li><a class="toctext" href="../../Demos/CurveResolution/">Curve Resolution</a></li></ul></li><li><span class="toctext">API</span><ul><li><a class="toctext" href="../Preprocess/">Preprocessing</a></li><li><a class="toctext" href="../Transformations/">Transformations/Pipelines</a></li><li><a class="toctext" href="../Sampling/">Sampling</a></li><li><a class="toctext" href="../Training/">Training</a></li><li><a class="toctext" href="../TimeSeries/">Time Series</a></li><li><a class="toctext" href="../RegressionModels/">Regression Models</a></li><li><a class="toctext" href="../regressMetrics/">Regression Metrics</a></li><li><a class="toctext" href="../ClassificationModels/">Classification Models</a></li><li><a class="toctext" href="../classMetrics/">Classification Metrics</a></li><li><a class="toctext" href="../Trees/">Tree Methods</a></li><li><a class="toctext" href="../Ensemble/">Ensemble Models</a></li><li><a class="toctext" href="../Clustering/">Clustering</a></li><li><a class="toctext" href="../MultiWay/">MultiWay</a></li><li><a class="toctext" href="../AnomalyDetection/">Anomaly Detection</a></li><li><a class="toctext" href="../CurveResolution/">Curve Resolution</a></li><li><a class="toctext" href="../Stats/">Stats.</a></li><li><a class="toctext" href="../Dists/">Distance Measures</a></li><li><a class="toctext" href="../PSO/">PSO</a></li><li><a class="toctext" href="../GeneticAlgorithms/">Genetic Algorithms</a></li></ul></li><li class="current"><a class="toctext" href>Full API</a><ul class="internal"></ul></li></ul></nav><article id="docs"><header><nav><ul><li><a href>Full API</a></li></ul><a class="edit-page" href="https://github.com/caseykneale/ChemometricsTools.jl/blob/master/docs/src/man/FullAPI.md"><span class="fa"></span> Edit on GitHub</a></nav><hr/><div id="topbar"><span>Full API</span><a class="fa fa-bars" href="#"></a></div></header><h1><a class="nav-anchor" id="API-1" href="#API-1">API</a></h1><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.BlandAltman-Tuple{Any,Any}" href="#ChemometricsTools.BlandAltman-Tuple{Any,Any}"><code>ChemometricsTools.BlandAltman</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">BlandAltman(Y1, Y2; Confidence = 1.96)</code></pre><p>Returns a Plot object of a Bland-Altman plot between vectors <code>Y1</code> and <code>Y2</code> with a confidence limit of <code>Confidence</code>.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.Bounds-Tuple{Any,Any,Any}" href="#ChemometricsTools.Bounds-Tuple{Any,Any,Any}"><code>ChemometricsTools.Bounds</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">Bounds(dims)</code></pre><p>Constructor for a Bounds object. Returns a bounds object with a lower bound of [lower...] and upper bound[upper...] with length of <code>dims</code>.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.Bounds-Tuple{Any}" href="#ChemometricsTools.Bounds-Tuple{Any}"><code>ChemometricsTools.Bounds</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">Bounds(dims)</code></pre><p>Default constructor for a Bounds object. Returns a bounds object with a lower bound of [0...] and upper bound[1...] with length of <code>dims</code>.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.CORAL-Tuple{Any,Any}" href="#ChemometricsTools.CORAL-Tuple{Any,Any}"><code>ChemometricsTools.CORAL</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">CORAL(X1, X2; lambda = 1.0)</code></pre><p>Performs CORAL to facilitate covariance based transfer from <code>X1</code> to <code>X2</code> with regularization parameter <code>lambda</code>. Returns a CORAL object.</p><p>Correlation Alignment for Unsupervised Domain Adaptation. Baochen Sun, Jiashi Feng, Kate Saenko. https://arxiv.org/abs/1612.01939</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.CORAL-Tuple{Any}" href="#ChemometricsTools.CORAL-Tuple{Any}"><code>ChemometricsTools.CORAL</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">(C::CORAL)(Z)</code></pre><p>Applies a the transform from a learned <code>CORAL</code> object to new data <code>Z</code>.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.CanonicalCorrelationAnalysis-Tuple{Any,Any}" href="#ChemometricsTools.CanonicalCorrelationAnalysis-Tuple{Any,Any}"><code>ChemometricsTools.CanonicalCorrelationAnalysis</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">CanonicalCorrelationAnalysis(A, B)</code></pre><p>Returns a CanonicalCorrelationAnalysis object which contains (U, V, r) from Arrays A and B. Currently Untested for correctness but should compute....</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.Center-Tuple{Any}" href="#ChemometricsTools.Center-Tuple{Any}"><code>ChemometricsTools.Center</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">(T::Center)(Z; inverse = false)</code></pre><p>Centers data in array <code>Z</code> column-wise according to learned mean centers in Center object <code>T</code>.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.CenterScale-Tuple{Any}" href="#ChemometricsTools.CenterScale-Tuple{Any}"><code>ChemometricsTools.CenterScale</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">(T::CenterScale)(Z; inverse = false)</code></pre><p>Centers and Scales data in array <code>Z</code> column-wise according to learned measures of central tendancy in Scale object <code>T</code>.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.ClassicLeastSquares-Tuple{Any,Any}" href="#ChemometricsTools.ClassicLeastSquares-Tuple{Any,Any}"><code>ChemometricsTools.ClassicLeastSquares</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">ClassicLeastSquares( X, Y; Bias = false )</code></pre><p>Makes a ClassicLeastSquares regression model of the form <code>Y</code> = A<code>X</code> with or without a <code>Bias</code> term. Returns a CLS object.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.ClassicLeastSquares-Tuple{Any}" href="#ChemometricsTools.ClassicLeastSquares-Tuple{Any}"><code>ChemometricsTools.ClassicLeastSquares</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">(M::ClassicLeastSquares)(X)</code></pre><p>Makes an inference from <code>X</code> using a ClassicLeastSquares object.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.GaussianBand" href="#ChemometricsTools.GaussianBand"><code>ChemometricsTools.GaussianBand</code></a> — <span class="docstring-category">Type</span>.</div><div><div><pre><code class="language-julia">GaussianBand(sigma,amplitude,center)</code></pre><p>Constructs a Gaussian kernel generator.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.GaussianBand-Tuple{Float64}" href="#ChemometricsTools.GaussianBand-Tuple{Float64}"><code>ChemometricsTools.GaussianBand</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">(B::GaussianBand)(X::Float64)</code></pre><p>Returns the scalar probability associated with a GaussianBand object (kernel) at a location in space(<code>X</code>).</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.KFoldsValidation-Tuple{Int64,Any,Any}" href="#ChemometricsTools.KFoldsValidation-Tuple{Int64,Any,Any}"><code>ChemometricsTools.KFoldsValidation</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">KFoldsValidation(K::Int, x, y)</code></pre><p>Returns a KFoldsValidation iterator with <code>K</code> folds. Because it&#39;s an iterator it can be used in for loops, see the tutorials for pragmatic examples. The iterator returns a 2-Tuple of 2-Tuples which have the  following form: <code>((TrainX,TrainY),(ValidateX,ValidateY)</code>.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.LDA-Tuple{Any,Any}" href="#ChemometricsTools.LDA-Tuple{Any,Any}"><code>ChemometricsTools.LDA</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">LDA(X, Y; Factors = 1)</code></pre><p>Compute&#39;s a LinearDiscriminantAnalysis transform from <code>x</code> with a user specified number of latent variables(<code>Factors</code>). Returns an LDA object.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.LDA-Tuple{Any}" href="#ChemometricsTools.LDA-Tuple{Any}"><code>ChemometricsTools.LDA</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">( model::LDA )( Z; Factors = length(model.Values) )</code></pre><p>Calling a LDA object on new data brings the new data <code>Z</code> into the LDA basis.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.LSSVM-Tuple{Any,Any,Any}" href="#ChemometricsTools.LSSVM-Tuple{Any,Any,Any}"><code>ChemometricsTools.LSSVM</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">LSSVM( X, Y, Penalty; KernelParameter = 0.0, KernelType = &quot;linear&quot; )</code></pre><p>Makes a LSSVM model of the form <code>Y</code> = A<code>K</code> with a bias term using a user specified Kernel(&quot;linear&quot;, or &quot;gaussian&quot;) and has an L2 <code>Penalty</code>. Returns a LSSVM Wrapper for a CLS object.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.LSSVM-Tuple{Any}" href="#ChemometricsTools.LSSVM-Tuple{Any}"><code>ChemometricsTools.LSSVM</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">(M::LSSVM)(X)</code></pre><p>Makes an inference from <code>X</code> using a LSSVM object.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.LorentzianBand" href="#ChemometricsTools.LorentzianBand"><code>ChemometricsTools.LorentzianBand</code></a> — <span class="docstring-category">Type</span>.</div><div><div><pre><code class="language-julia">LorentzianBand(gamma,amplitude,center)</code></pre><p>Constructs a Lorentzian kernel generator.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.LorentzianBand-Tuple{Float64}" href="#ChemometricsTools.LorentzianBand-Tuple{Float64}"><code>ChemometricsTools.LorentzianBand</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">(B::LorentzianBand)(X::Float64)</code></pre><p>Returns the probability associated with a LorentzianBand object (kernel) at a location in space(<code>X</code>).</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.MultiCenter" href="#ChemometricsTools.MultiCenter"><code>ChemometricsTools.MultiCenter</code></a> — <span class="docstring-category">Type</span>.</div><div><div><pre><code class="language-julia">MultiCenter(Z, mode = 1)</code></pre><p>Acquires the mean of the specified mode in <code>Z</code> and returns a transform that will remove those means from any future data.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.MultiCenter-Tuple{Any}" href="#ChemometricsTools.MultiCenter-Tuple{Any}"><code>ChemometricsTools.MultiCenter</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">(T::MultiCenter)(Z; inverse = false)</code></pre><p>Centers data in Tensor <code>Z</code> mode-wise according to learned centers in MultiCenter object <code>T</code>.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.MultiScale" href="#ChemometricsTools.MultiScale"><code>ChemometricsTools.MultiScale</code></a> — <span class="docstring-category">Type</span>.</div><div><div><pre><code class="language-julia">MultiScale(Z, mode = 1)</code></pre><p>Acquires the standard deviations of the specified mode in <code>Z</code> and returns a transform that will scale by those standard deviations from any future data.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.MultiScale-Tuple{Any}" href="#ChemometricsTools.MultiScale-Tuple{Any}"><code>ChemometricsTools.MultiScale</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">(T::MultiScale)(Z; inverse = false)</code></pre><p>Scales data in Tensor <code>Z</code> mode-wise according to learned standard deviations in MultiScale object <code>T</code>.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.MultiplicativeScatterCorrection-Tuple{Any}" href="#ChemometricsTools.MultiplicativeScatterCorrection-Tuple{Any}"><code>ChemometricsTools.MultiplicativeScatterCorrection</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">(T::MultiplicativeScatterCorrection)(Z)</code></pre><p>Applies MultiplicativeScatterCorrection from a stored object <code>T</code> to Array <code>Z</code>.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.OrthogonalSignalCorrection-Tuple{Any,Any}" href="#ChemometricsTools.OrthogonalSignalCorrection-Tuple{Any,Any}"><code>ChemometricsTools.OrthogonalSignalCorrection</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">OrthogonalSignalCorrection(X, Y; Factors = 1)</code></pre><p>Performs Thomas Fearn&#39;s Orthogonal Signal Correction to an endogenous <code>X</code> and exogenous <code>Y</code>. The number of <code>Factors</code> are the number of orthogonal components to be removed from <code>X</code>. This function returns an OSC object.</p><p>Tom Fearn. On orthogonal signal correction. Chemometrics and Intelligent Laboratory Systems. Volume 50, Issue 1, 2000, Pages 47-52.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.OrthogonalSignalCorrection-Tuple{Any}" href="#ChemometricsTools.OrthogonalSignalCorrection-Tuple{Any}"><code>ChemometricsTools.OrthogonalSignalCorrection</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">(OSC::OrthogonalSignalCorrection)(Z; Factors = 2)</code></pre><p>Applies a the transform from a learned orthogonal signal correction object <code>OSC</code> to new data <code>Z</code>.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.PCA-Tuple{Any}" href="#ChemometricsTools.PCA-Tuple{Any}"><code>ChemometricsTools.PCA</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">PCA(X; Factors = minimum(size(X)) - 1)</code></pre><p>Compute&#39;s a PCA from <code>x</code> using LinearAlgebra&#39;s SVD algorithm with a user specified number of latent variables(<code>Factors</code>). Returns a PCA object.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.PCA-Tuple{Array}" href="#ChemometricsTools.PCA-Tuple{Array}"><code>ChemometricsTools.PCA</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">(T::PCA)(Z::Array; Factors = length(T.Values), inverse = false)</code></pre><p>Calling a PCA object on new data brings the new data <code>Z</code> into or out of (<code>inverse</code> = true) the PCA basis.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.PartialLeastSquares-Tuple{Any,Any}" href="#ChemometricsTools.PartialLeastSquares-Tuple{Any,Any}"><code>ChemometricsTools.PartialLeastSquares</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">PartialLeastSquares( X, Y; Factors = minimum(size(X)) - 2, tolerance = 1e-8, maxiters = 200 )</code></pre><p>Returns a PartialLeastSquares regression model object from arrays <code>X</code> and <code>Y</code>.</p><ol><li>PARTIAL LEAST-SQUARES REGRESSION: A TUTORIAL PAUL GELADI and BRUCE R.KOWALSKI. Analytica Chimica Acta, 186, (1986) PARTIAL LEAST-SQUARES REGRESSION:</li><li>Martens H., NÊs T. Multivariate Calibration. Wiley: New York, 1989.</li><li>Re-interpretation of NIPALS results solves PLSR inconsistency problem. Rolf Ergon. Published in Journal of Chemometrics 2009; Vol. 23/1: 72-75</li></ol></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.PartialLeastSquares-Tuple{Any}" href="#ChemometricsTools.PartialLeastSquares-Tuple{Any}"><code>ChemometricsTools.PartialLeastSquares</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">(M::PartialLeastSquares)</code></pre><p>Makes an inference from <code>X</code> using a PartialLeastSquares object.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.Particle-Tuple{Any,Any}" href="#ChemometricsTools.Particle-Tuple{Any,Any}"><code>ChemometricsTools.Particle</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">Particle(ProblemBounds, VelocityBounds)</code></pre><p>Default constructor for a Particle object. It creates a random unformly distributed particle within the specified <code>ProblemBounds</code>, and limits it&#39;s velocity to the specified <code>VelocityBounds</code>.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.PrincipalComponentRegression-Tuple{Any}" href="#ChemometricsTools.PrincipalComponentRegression-Tuple{Any}"><code>ChemometricsTools.PrincipalComponentRegression</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">(M::PrincipalComponentRegression)( X )</code></pre><p>Makes an inference from <code>X</code> using a PrincipalComponentRegression object.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.PrincipalComponentRegression-Tuple{PCA,Any}" href="#ChemometricsTools.PrincipalComponentRegression-Tuple{PCA,Any}"><code>ChemometricsTools.PrincipalComponentRegression</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">PrincipalComponentRegression(PCAObject, Y )</code></pre><p>Makes a PrincipalComponentRegression model object from a PCA Object and property value <code>Y</code>.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.QQ-Tuple{Any,Any}" href="#ChemometricsTools.QQ-Tuple{Any,Any}"><code>ChemometricsTools.QQ</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">QQ( Y1, Y2; Quantiles = collect( 1 : 99 ) ./ 100 )</code></pre><p>Returns a Plot object of a Quantile-Quantile plot between vectors <code>Y1</code> and <code>Y2</code> at the desired <code>Quantiles</code>.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.QuantileTrim" href="#ChemometricsTools.QuantileTrim"><code>ChemometricsTools.QuantileTrim</code></a> — <span class="docstring-category">Type</span>.</div><div><div><pre><code class="language-julia">(T::QuantileTrim)(X, inverse = false)</code></pre><p>Trims data in array <code>X</code> columns wise according to learned quantiles in QuantileTrim object <code>T</code> This function does NOT have an inverse.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.QuantileTrim-Tuple{Any}" href="#ChemometricsTools.QuantileTrim-Tuple{Any}"><code>ChemometricsTools.QuantileTrim</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">QuantileTrim(Z; quantiles::Tuple{Float64,Float64} = (0.05, 0.95) )</code></pre><p>Trims values above or below the specified columnwise quantiles to the quantile values themselves.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.RangeNorm-Tuple{Any}" href="#ChemometricsTools.RangeNorm-Tuple{Any}"><code>ChemometricsTools.RangeNorm</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">(T::RangeNorm)(Z; inverse = false)</code></pre><p>Scales and shifts data in array <code>Z</code> column-wise according to learned min-maxes in RangeNorm object <code>T</code>.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.RidgeRegression-Tuple{Any,Any,Any}" href="#ChemometricsTools.RidgeRegression-Tuple{Any,Any,Any}"><code>ChemometricsTools.RidgeRegression</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">RidgeRegression( X, Y, Penalty; Bias = false )</code></pre><p>Makes a RidgeRegression model of the form <code>Y</code> = A<code>X</code> with or without a <code>Bias</code> term and has an L2 <code>Penalty</code>. Returns a CLS object.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.RidgeRegression-Tuple{Any}" href="#ChemometricsTools.RidgeRegression-Tuple{Any}"><code>ChemometricsTools.RidgeRegression</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">(M::RidgeRegression)(X)</code></pre><p>Makes an inference from <code>X</code> using a RidgeRegression object which wraps a ClassicLeastSquares object.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.RollingWindow-Tuple{Int64,Int64,Int64}" href="#ChemometricsTools.RollingWindow-Tuple{Int64,Int64,Int64}"><code>ChemometricsTools.RollingWindow</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">RollingWindow(samples::Int,windowsize::Int,skip::Int)</code></pre><p>Creates a RollingWindow iterator from a number of <code>samples</code> and a static <code>windowsize</code> where every iteration <code>skip</code> steps are skipped. The iterator can be used in for loops to iteratively return indices of a dynamic rolling window.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.RollingWindow-Tuple{Int64,Int64}" href="#ChemometricsTools.RollingWindow-Tuple{Int64,Int64}"><code>ChemometricsTools.RollingWindow</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">RollingWindow(samples::Int,windowsize::Int)</code></pre><p>Creates a RollingWindow iterator from a number of <code>samples</code> and a static <code>windowsize</code>. The iterator can be used in for loops to iteratively return indices of a dynamic rolling window.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.RunningMean-Tuple{Any}" href="#ChemometricsTools.RunningMean-Tuple{Any}"><code>ChemometricsTools.RunningMean</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">RunningMean(x)</code></pre><p>Constructs a running mean object with an initial scalar value of <code>x</code>.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.RunningVar-Tuple{Any}" href="#ChemometricsTools.RunningVar-Tuple{Any}"><code>ChemometricsTools.RunningVar</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">RunningVar(x)</code></pre><p>Constructs a RunningVar object with an initial scalar value of <code>x</code>. <em>Note: RunningVar objects implicitly calculate the running mean.</em></p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.Scale-Tuple{Any}" href="#ChemometricsTools.Scale-Tuple{Any}"><code>ChemometricsTools.Scale</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">(T::Scale)(Z; inverse = false)</code></pre><p>Scales data in array <code>Z</code> column-wise according to learned standard deviations in Scale object <code>T</code>.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.TransferByOrthogonalProjection-Tuple{Any,Any}" href="#ChemometricsTools.TransferByOrthogonalProjection-Tuple{Any,Any}"><code>ChemometricsTools.TransferByOrthogonalProjection</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">TransferByOrthogonalProjection(X1, X2; Factors = 1)</code></pre><p>Performs Thomas Fearns Transfer By Orthogonal Projection to facilitate transfer from <code>X1</code> to <code>X2</code>. Returns a TransferByOrthogonalProjection object.</p><p>Anne Andrew, Tom Fearn. Transfer by orthogonal projection: making near-infrared calibrations robust to between-instrument variation. Chemometrics and Intelligent Laboratory Systems. Volume 72, Issue 1, 2004, Pages 51-56,</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.TransferByOrthogonalProjection-Tuple{Any}" href="#ChemometricsTools.TransferByOrthogonalProjection-Tuple{Any}"><code>ChemometricsTools.TransferByOrthogonalProjection</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">(TbOP::TransferByOrthogonalProjection)(X1; Factors = TbOP.Factors)</code></pre><p>Applies a the transform from a learned transfer by orthogonal projection object <code>TbOP</code> to new data <code>X1</code>.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.Universe-Tuple" href="#ChemometricsTools.Universe-Tuple"><code>ChemometricsTools.Universe</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">(U::Universe)(Band...)</code></pre><p>A Universe objects internal &quot;spectra&quot; can be updated to include the additive contribution of many Band-like objects.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.Universe-Tuple{Any,Any}" href="#ChemometricsTools.Universe-Tuple{Any,Any}"><code>ChemometricsTools.Universe</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">Universe(mini, maxi; width = nothing, bins = nothing)</code></pre><p>Creates a 1-D discretized segment that starts at mini and ends at maxi. The width of the bins for the discretization can either be provided or inferred from the number of bins. Returns a Universe object.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.Universe-Tuple{Union{GaussianBand, LorentzianBand}}" href="#ChemometricsTools.Universe-Tuple{Union{GaussianBand, LorentzianBand}}"><code>ChemometricsTools.Universe</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">(U::Universe)(Band::Union{ GaussianBand, LorentzianBand})</code></pre><p>A Universe objects internal &quot;spectra&quot; can be updated to include the additive contribution of any Band-like object.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.ALSSmoother-Tuple{Any}" href="#ChemometricsTools.ALSSmoother-Tuple{Any}"><code>ChemometricsTools.ALSSmoother</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">ALSSmoother(X; lambda = 100, p = 0.001, maxiters = 10)</code></pre><p>Applies an assymetric least squares smoothing function to a 2-Array <code>X</code>. The <code>lambda</code>, <code>p</code>, and <code>maxiters</code> parameters control the smoothness. See the reference below for more information.</p><p>Paul H. C. Eilers, Hans F.M. Boelens. Baseline Correction with Asymmetric Least Squares Smoothing.  2005</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.AssessHealth-Tuple{Any}" href="#ChemometricsTools.AssessHealth-Tuple{Any}"><code>ChemometricsTools.AssessHealth</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">AssessHealth( X )</code></pre><p>Returns a somewhat detailed Dict containing information about the &#39;health&#39; of a dataset. What is included is the following:     - PercentMissing: percent of missing entries (includes nothing, inf / nan) in the dataset     - EmptyColumns: the columns which have only 1 value     - RankEstimate: An estimate of the rank of X     - (optional)Duplicates: returns the rows of duplicate observations</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.ClassificationTree-Tuple{Any,Any}" href="#ChemometricsTools.ClassificationTree-Tuple{Any,Any}"><code>ChemometricsTools.ClassificationTree</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">ClassificationTree(x, y; gainfn = entropy, maxdepth = 4, minbranchsize = 3)</code></pre><p>Builds a CART object using either gini or entropy as a partioning method. Y must be a one hot encoded 2-Array. Predictions can be formed by calling the following function from the CART object: (M::CART)(x).</p><p>*Note: this is a purely nonrecursive decision tree. The julia compiler doesn&#39;t like storing structs of nested things. I wrote it the recursive way in the past and it was quite slow, I think this is true also of interpretted languages like R/Python...So here it is, nonrecursive tree&#39;s!</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.ColdToHot-Tuple{Any,ChemometricsTools.ClassificationLabel}" href="#ChemometricsTools.ColdToHot-Tuple{Any,ChemometricsTools.ClassificationLabel}"><code>ChemometricsTools.ColdToHot</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">ColdToHot(Y, Schema::ClassificationLabel)</code></pre><p>Turns a cold encoded <code>Y</code> vector into a one hot encoded array.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.DirectStandardization-Tuple{Any,Any}" href="#ChemometricsTools.DirectStandardization-Tuple{Any,Any}"><code>ChemometricsTools.DirectStandardization</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">DirectStandardization(InstrumentX1, InstrumentX2; Factors = minimum(collect(size(InstrumentX1))) - 1)</code></pre><p>Makes a DirectStandardization object to facilitate the transfer from Instrument #2 to Instrument #1 . The returned object can be used to transfer unseen data to the approximated space of instrument 1. The number of <code>Factors</code> used are those from the internal orthogonal basis.</p><p>Yongdong Wang and Bruce R. Kowalski, &quot;Calibration Transfer and Measurement Stability of Near-Infrared Spectrometers,&quot; Appl. Spectrosc. 46, 764-771 (1992)</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.EWMA-Tuple{Array,Float64}" href="#ChemometricsTools.EWMA-Tuple{Array,Float64}"><code>ChemometricsTools.EWMA</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">EWMA(Initial::Float64, Lambda::Float64) = ewma(Lambda, Initial, Initial, RunningVar(Initial))</code></pre><p>Constructs an exponentially weighted moving average object from an vector of scalar property values <code>Initial</code> and the decay parameter <code>Lambda</code>. This computes the running statistcs neccesary for creating the EWMA model using the interval provided and updates the center value to the mean of the provided values.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.EWMA-Tuple{Float64,Float64}" href="#ChemometricsTools.EWMA-Tuple{Float64,Float64}"><code>ChemometricsTools.EWMA</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">EWMA(Initial::Float64, Lambda::Float64) = ewma(Lambda, Initial, Initial, RunningVar(Initial))</code></pre><p>Constructs an exponentially weighted moving average object from an initial scalar property value <code>Initial</code> and the decay parameter <code>Lambda</code>. This defaults the center value to be the initial value.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.EmpiricalQuantiles-Tuple{Any,Any}" href="#ChemometricsTools.EmpiricalQuantiles-Tuple{Any,Any}"><code>ChemometricsTools.EmpiricalQuantiles</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">EmpiricalQuantiles(X, quantiles)</code></pre><p>Finds the column-wise <code>quantiles</code> of 2-Array <code>X</code> and returns them in a 2-Array of size <code>quantiles</code> by <code>variables</code>. *Note: This copies the array... Use a subset if memory is the concern. *</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.ExplainedVariance-Tuple{LDA}" href="#ChemometricsTools.ExplainedVariance-Tuple{LDA}"><code>ChemometricsTools.ExplainedVariance</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">ExplainedVariance(lda::LDA)</code></pre><p>Calculates the explained variance of each singular value in an LDA object.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.ExplainedVariance-Tuple{PCA}" href="#ChemometricsTools.ExplainedVariance-Tuple{PCA}"><code>ChemometricsTools.ExplainedVariance</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">ExplainedVariance(PCA::PCA)</code></pre><p>Calculates the explained variance of each singular value in a pca object.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.ExtremeLearningMachine" href="#ChemometricsTools.ExtremeLearningMachine"><code>ChemometricsTools.ExtremeLearningMachine</code></a> — <span class="docstring-category">Function</span>.</div><div><div><pre><code class="language-julia">ExtremeLearningMachine(X, Y, ReservoirSize = 10; ActivationFn = sigmoid)</code></pre><p>Returns a ELM regression model object from arrays <code>X</code> and <code>Y</code>, with a user specified <code>ReservoirSize</code> and <code>ActivationFn</code>.</p><p>Extreme learning machine: a new learning scheme of feedforward neural networks. Guang-Bin Huang ; Qin-Yu Zhu ; Chee-Kheong Siew. 	2004 IEEE International Joint...</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.FirstDerivative-Tuple{Any}" href="#ChemometricsTools.FirstDerivative-Tuple{Any}"><code>ChemometricsTools.FirstDerivative</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">FirstDerivative(X)</code></pre><p>Uses the finite difference method to compute the first derivative for every row in <code>X</code>. <em>Note: This operation results in the loss of a column dimension.</em></p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.FractionalDerivative" href="#ChemometricsTools.FractionalDerivative"><code>ChemometricsTools.FractionalDerivative</code></a> — <span class="docstring-category">Function</span>.</div><div><div><pre><code class="language-julia">FractionalDerivative(Y, X = 1 : length(Y); Order = 0.5)</code></pre><p>Calculates the Grunwald-Leitnikov fractional order derivative on every row of Array Y. Array <code>X</code> is a vector that has the spacing between column-wise entries in <code>Y</code>. <code>X</code> can be a scalar if that is constant (common in spectroscopy). <code>Order</code> is the fractional order of the derivative. <em>Note: This operation results in the loss of a column dimension.</em></p><p>The Fractional Calculus, by Oldham, K.; and Spanier, J. Hardcover: 234 pages. Publisher: Academic Press, 1974. ISBN 0-12-525550-0</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.HLDA-Tuple{Any,Any}" href="#ChemometricsTools.HLDA-Tuple{Any,Any}"><code>ChemometricsTools.HLDA</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">HLDA(X, YHOT; K = 1, Factors = 1)</code></pre><p>Compute&#39;s a Hierarchical LinearDiscriminantAnalysis transform from <code>x</code> with a user specified number of latent variables(<code>Factors</code>). The adjacency matrices are created from <code>K</code> nearest neighbors.</p><p>Returns an LDA object. <em>Note: this can be used with any other LDA functions such as Gaussian discriminants or explained variance</em>.</p><p>Lu D, Ding C, Xu J, Wang S. Hierarchical Discriminant Analysis. Sensors (Basel). 2018 Jan 18;18(1). pii: E279. doi: 10.3390/s18010279.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.HighestVote-Tuple{Any}" href="#ChemometricsTools.HighestVote-Tuple{Any}"><code>ChemometricsTools.HighestVote</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">HighestVote(yhat)</code></pre><p>Returns the column index for each row that has the highest value in one hot encoded <code>yhat</code>. Returns a one cold encoded vector.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.HighestVoteOneHot-Tuple{Any}" href="#ChemometricsTools.HighestVoteOneHot-Tuple{Any}"><code>ChemometricsTools.HighestVoteOneHot</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">HighestVoteOneHot(yhat)</code></pre><p>Turns the highest column-wise value to a 1 and the others to zeros per row in a one hot encoded <code>yhat</code>. Returns a one cold encoded vector.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.HotToCold-Tuple{Any,ChemometricsTools.ClassificationLabel}" href="#ChemometricsTools.HotToCold-Tuple{Any,ChemometricsTools.ClassificationLabel}"><code>ChemometricsTools.HotToCold</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">HotToCold(Y, Schema::ClassificationLabel)</code></pre><p>Turns a one hot encoded <code>Y</code> array into a cold encoded vector.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.IntervalOverlay-Tuple{Any,Any,Any}" href="#ChemometricsTools.IntervalOverlay-Tuple{Any,Any,Any}"><code>ChemometricsTools.IntervalOverlay</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">IntervalOverlay(Spectra, Intervals, Err)</code></pre><p>Displays the relative error(<code>Err</code>) of each <code>interval</code> ontop of a <code>Spectra</code>.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.IsColdEncoded-Tuple{Any}" href="#ChemometricsTools.IsColdEncoded-Tuple{Any}"><code>ChemometricsTools.IsColdEncoded</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">IsColdEncoded(Y)</code></pre><p>Returns a boolean true if the array Y is cold encoded, and false if not.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.KennardStone-Tuple{Any,Any}" href="#ChemometricsTools.KennardStone-Tuple{Any,Any}"><code>ChemometricsTools.KennardStone</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">KennardStone(X, TrainSamples; distance = &quot;euclidean&quot;)</code></pre><p>Returns the indices of the Kennard-Stone sampled exemplars (E), and those not sampled (O) as a 2-Tuple (E, O).</p><p>R. W. Kennard &amp; L. A. Stone (1969) Computer Aided Design of Experiments, Technometrics, 111, 137-148, DOI: 10.1080/00401706.1969.10490666</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.KernelRidgeRegression-Tuple{Any,Any,Any}" href="#ChemometricsTools.KernelRidgeRegression-Tuple{Any,Any,Any}"><code>ChemometricsTools.KernelRidgeRegression</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">KernelRidgeRegression( X, Y, Penalty; KernelParameter = 0.0, KernelType = &quot;linear&quot; )</code></pre><p>Makes a KernelRidgeRegression model of the form <code>Y</code> = A<code>K</code> using a user specified Kernel(&quot;Linear&quot;, or &quot;Guassian&quot;) and has an L2 <code>Penalty</code>. Returns a KRR Wrapper for a CLS object.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.LabelEncoding-Tuple{Any}" href="#ChemometricsTools.LabelEncoding-Tuple{Any}"><code>ChemometricsTools.LabelEncoding</code></a> — <span class="docstring-category">Method</span>.</div><div><div><p>&quot;     LabelEncoding(HotOrCold)</p><p>Determines if an Array, <code>Y</code>, is one hot encoded, or cold encoded by it&#39;s dimensions. Returns a ClassificationLabel object/schema to convert between the formats.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.LeaveOneOut-Tuple{Any,Any}" href="#ChemometricsTools.LeaveOneOut-Tuple{Any,Any}"><code>ChemometricsTools.LeaveOneOut</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">LeaveOneOut(x, y)</code></pre><p>Returns a KFoldsValidation iterator with leave one out folds. Because it&#39;s an iterator it can be used in for loops, see the tutorials for pragmatic examples. The iterator returns a 2-Tuple of 2-Tuples which have the  following form: <code>((TrainX,TrainY),(ValidateX,ValidateY)</code>.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.Lifeform-Tuple{Any,Any,Any}" href="#ChemometricsTools.Lifeform-Tuple{Any,Any,Any}"><code>ChemometricsTools.Lifeform</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">Lifeform(size, onlikelihood, initialscore)</code></pre><p>Constructor for a BinaryLifeForm struct. Binary life forms are basically wrappers for a binary vector, which has a likelihood for being 1(<code>onlikelihood</code>). Each life form also has a <code>score</code> based on it&#39;s &quot;fitness&quot;. So the GA&#39;s in this package can be used to minimize or maximize this is an open parameter, but Inf/-Inf is a good <code>initialscore</code>.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.Limits-Tuple{ChemometricsTools.ewma}" href="#ChemometricsTools.Limits-Tuple{ChemometricsTools.ewma}"><code>ChemometricsTools.Limits</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">Limits(P::ewma; k = 3.0)</code></pre><p>This function returns the upper and lower control limits with a <code>k</code> span of variance for an EWMA object <code>P</code>.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.Logit-Tuple{Any}" href="#ChemometricsTools.Logit-Tuple{Any}"><code>ChemometricsTools.Logit</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">Logit(Z; inverse = false)</code></pre><p>Logit transforms (<code>ln( X / (1 - X) ))</code>) every element in <code>Z</code>. The inverse may also be applied. <em>Warning: This can return Infs and NaNs if elements of Z are not suited to the transform</em></p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.MAE-Tuple{Any,Any}" href="#ChemometricsTools.MAE-Tuple{Any,Any}"><code>ChemometricsTools.MAE</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">MAE( y, yhat )</code></pre><p>Calculates Mean Average Error from vectors <code>Y</code> and <code>YHat</code></p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.MAPE-Tuple{Any,Any}" href="#ChemometricsTools.MAPE-Tuple{Any,Any}"><code>ChemometricsTools.MAPE</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">MAPE( y, yhat )</code></pre><p>Calculates Mean Average Percent Error from vectors <code>Y</code> and <code>YHat</code></p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.ME-Tuple{Any,Any}" href="#ChemometricsTools.ME-Tuple{Any,Any}"><code>ChemometricsTools.ME</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">ME( y, yhat )</code></pre><p>Calculates Mean Error from vectors <code>Y</code> and <code>YHat</code>.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.MSE-Tuple{Any,Any}" href="#ChemometricsTools.MSE-Tuple{Any,Any}"><code>ChemometricsTools.MSE</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">MSE( y, yhat )</code></pre><p>Calculates Mean Squared Error from vectors <code>Y</code> and <code>YHat</code></p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.Mean-Tuple{RunningMean}" href="#ChemometricsTools.Mean-Tuple{RunningMean}"><code>ChemometricsTools.Mean</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">Mean(rv::RunningMean)</code></pre><p>Returns the current mean inside of a RunningMean object.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.Mean-Tuple{RunningVar}" href="#ChemometricsTools.Mean-Tuple{RunningVar}"><code>ChemometricsTools.Mean</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">Mean(rv::RunningVar)</code></pre><p>Returns the current mean inside of a RunningVar object.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.MultiNorm-Tuple{Any}" href="#ChemometricsTools.MultiNorm-Tuple{Any}"><code>ChemometricsTools.MultiNorm</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">MultiNorm(T)</code></pre><p>Computes the equivalent of the Froebinius norm on a tensor <code>T</code>. Returns a scalar.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.MultiPCA-Tuple{Any}" href="#ChemometricsTools.MultiPCA-Tuple{Any}"><code>ChemometricsTools.MultiPCA</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">MultiPCA(X; Factors = 2)</code></pre><p>Performs multiway PCA aka Higher Order SVD aka Tucker, etc. The number of factors decomposed can be a scalar(repeated across all modes) or a vector/tuple for each mode.</p><p>Returns a tuple of (Core Tensor, Basis Tensors)</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.MulticlassStats-Tuple{Any,Any,Any}" href="#ChemometricsTools.MulticlassStats-Tuple{Any,Any,Any}"><code>ChemometricsTools.MulticlassStats</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">MulticlassStats(Y, GT, schema; Microaverage = true)</code></pre><p>Calculates many essential classification statistics based on predicted values <code>Y</code>, and ground truth values <code>GT</code>, using the encoding <code>schema</code>. Returns a tuple whose first entry is a dictionary of averaged statistics, and whose second entry is a dictionary of the form &quot;Class&quot; =&gt; Statistics Dictionary ...</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.MulticlassThreshold-Tuple{Any}" href="#ChemometricsTools.MulticlassThreshold-Tuple{Any}"><code>ChemometricsTools.MulticlassThreshold</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">MulticlassThreshold(yhat; level = 0.5)</code></pre><p>Effectively does the same thing as Threshold() but per-row across columns.</p><p><em>Warning this function can allow for no class assignments. HighestVote is preferred</em></p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.Mutate" href="#ChemometricsTools.Mutate"><code>ChemometricsTools.Mutate</code></a> — <span class="docstring-category">Function</span>.</div><div><div><pre><code class="language-julia">Mutate( L::BinaryLifeform, amount = 0.05 )</code></pre><p>Assesses each element in the gene vector (inside of <code>L</code>). If a randomly drawn value has a binomial probability of <code>amount</code> the element is mutated.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.OneHotOdds-Tuple{Any}" href="#ChemometricsTools.OneHotOdds-Tuple{Any}"><code>ChemometricsTools.OneHotOdds</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">OneHotOdds(Y)</code></pre><p>Calculates the odds of a one-hot formatted probability matrix. Returns a tuple.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.PCA_NIPALS-Tuple{Any}" href="#ChemometricsTools.PCA_NIPALS-Tuple{Any}"><code>ChemometricsTools.PCA_NIPALS</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">PCA_NIPALS(X; Factors = minimum(size(X)) - 1, tolerance = 1e-7, maxiters = 200)</code></pre><p>Compute&#39;s a PCA from <code>x</code> using the NIPALS algorithm with a user specified number of latent variables(<code>Factors</code>). The tolerance is the minimum change in the F norm before ceasing execution. Returns a PCA object.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.PSO-Tuple{Any,Bounds,Bounds,Int64}" href="#ChemometricsTools.PSO-Tuple{Any,Bounds,Bounds,Int64}"><code>ChemometricsTools.PSO</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">PSO(fn, Bounds, VelRange, Particles; tolerance = 1e-6, maxiters = 1000, InertialDecay = 0.5, PersonalWeight = 0.5, GlobalWeight = 0.5, InternalParams = nothing)</code></pre><p>Minimizes function <code>fn</code> with-in the user specified <code>Bounds</code> via a Particle Swarm Optimizer. The particle velocities are limitted to the <code>VelRange</code>. The number of particles are defined by the <code>Particles</code> parameter.</p><p>Returns a Tuple of the following form: ( GlobalBestPos, GlobalBestScore, P ) Where P is an array of the particles used in the optimization.</p><p>*Note: if the optimization function requires an additional constant parameter, please pass that parameter to InternalParams. This will only work if the optimized parameter(o) and constant parameter(c) for the function of interest has the following format: F(o,c) *</p><p>Kennedy, J.; Eberhart, R. (1995). Particle Swarm Optimization. Proceedings of IEEE International Conference on Neural Networks. IV. pp. 1942–1948. doi:10.1109/ICNN.1995.488968</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.PearsonCorrelationCoefficient-Tuple{Any,Any}" href="#ChemometricsTools.PearsonCorrelationCoefficient-Tuple{Any,Any}"><code>ChemometricsTools.PearsonCorrelationCoefficient</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">PearsonCorrelationCoefficient( y, yhat )</code></pre><p>Calculates The Pearson Correlation Coefficient from vectors <code>Y</code> and <code>YHat</code></p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.PercentRMSE-Tuple{Any,Any}" href="#ChemometricsTools.PercentRMSE-Tuple{Any,Any}"><code>ChemometricsTools.PercentRMSE</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">PercentRMSE( y, yhat )</code></pre><p>Calculates Percent Root Mean Squared Error from vectors <code>Y</code> and <code>YHat</code></p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.PerfectSmoother-Tuple{Any}" href="#ChemometricsTools.PerfectSmoother-Tuple{Any}"><code>ChemometricsTools.PerfectSmoother</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">PerfectSmoother(X; lambda = 100)</code></pre><p>Applies an assymetric least squares smoothing function to a a 2-Array <code>X</code>. The <code>lambda</code> parameter controls the smoothness. See the reference below for more information.</p><p>Paul H. C. Eilers. &quot;A Perfect Smoother&quot;. Analytical Chemistry, 2003, 75 (14), pp 3631–3636.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.Pipeline-Tuple{Any,Vararg{Any,N} where N}" href="#ChemometricsTools.Pipeline-Tuple{Any,Vararg{Any,N} where N}"><code>ChemometricsTools.Pipeline</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">Pipeline( X, FnStack... )</code></pre><p>Construct a pipeline object from vector/tuple of <code>Transforms</code>. The Transforms vector are effectively a vector of functions which transform data.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.Pipeline-Tuple{Any}" href="#ChemometricsTools.Pipeline-Tuple{Any}"><code>ChemometricsTools.Pipeline</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">Pipeline(Transforms)</code></pre><p>Constructs a transformation pipeline from vector/tuple of <code>Transforms</code>. The Transforms vector are effectively a vector of functions which transform data.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.PipelineInPlace-Tuple{Any,Vararg{Any,N} where N}" href="#ChemometricsTools.PipelineInPlace-Tuple{Any,Vararg{Any,N} where N}"><code>ChemometricsTools.PipelineInPlace</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">PipelineInPlace( X, FnStack...)</code></pre><p>Construct a pipeline object from vector/tuple of <code>Transforms</code>. The Transforms vector are effectively a vector of functions which transform data. This function makes &quot;inplace&quot; changes to the Array <code>X</code> as though it has been sent through the pipeline. This is more efficient if memory is a concern, but can irreversibly transform data in memory depending on the transforms in the pipeline.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.RAFFT-Tuple{Any,Any}" href="#ChemometricsTools.RAFFT-Tuple{Any,Any}"><code>ChemometricsTools.RAFFT</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">RAFFT(raw, reference; maxlags::Int = 500, lookahead::Int = 1, minlength::Int = 20, mincorr::Float64 = 0.05)</code></pre><p>RAFFT corrects shifts in the <code>raw</code> spectral bands to be similar to those in a given <code>reference</code> spectra through the use of &quot;recursive alignment by FFT&quot;. It returns an array of corrected spectra/chromatograms. The number of maximum lags can be specified, the <code>lookahead</code> parameter ensures that additional recursive executions are performed so the first solution found is not preemptively accepted, the minimum segment length(<code>minlength</code>) can also be specified if FWHM are estimable, and the minimum cross correlation(<code>mincorr</code>) for a match can dictate whether peaks were found to align or not.</p><p><em>Note</em> This method works best with flat baselines because it repeats last known values when padding aligned spectra. It is highly efficient, and in my tests does a good job, but other methods definitely exist. Let me know if other peak Alignment methods are important for your work-flow, I&#39;ll see if I can implement them.</p><p>Application of Fast Fourier Transform Cross-Correlation for the Alignment of Large Chromatographic and Spectral Datasets Jason W. H. Wong, Caterina Durante, and, Hugh M. Cartwright. Analytical Chemistry 2005 77 (17), 5655-5661</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.RMSE-Tuple{Any,Any}" href="#ChemometricsTools.RMSE-Tuple{Any,Any}"><code>ChemometricsTools.RMSE</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">RMSE( y, yhat )</code></pre><p>Calculates Root Mean Squared Error from vectors <code>Y</code> and <code>YHat</code></p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.RSquare-Tuple{Any,Any}" href="#ChemometricsTools.RSquare-Tuple{Any,Any}"><code>ChemometricsTools.RSquare</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">RSquare( y, yhat )</code></pre><p>Calculates R^2 from <code>Y</code> and <code>YHat</code></p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.Remove!-Tuple{RunningMean,Any}" href="#ChemometricsTools.Remove!-Tuple{RunningMean,Any}"><code>ChemometricsTools.Remove!</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">Remove!(RM::RunningMean, x)</code></pre><p>Removes an observation(<code>x</code>) from a RunningMean object(<code>RM</code>) and reculates the mean in place.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.Remove-Tuple{RunningMean,Any}" href="#ChemometricsTools.Remove-Tuple{RunningMean,Any}"><code>ChemometricsTools.Remove</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">Remove!(RM::RunningMean, x)</code></pre><p>Removes an observation(<code>x</code>) from a RunningMean object(<code>RM</code>) and recuturns the new RunningMean object.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.SSE-Tuple{Any,Any}" href="#ChemometricsTools.SSE-Tuple{Any,Any}"><code>ChemometricsTools.SSE</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">SSE( y, yhat )</code></pre><p>Calculates Sum of Squared Errors from vectors <code>Y</code> and <code>YHat</code></p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.SSReg-Tuple{Any,Any}" href="#ChemometricsTools.SSReg-Tuple{Any,Any}"><code>ChemometricsTools.SSReg</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">SSReg( y, yhat )</code></pre><p>Calculates Sum of Squared Deviations due to Regression from vectors <code>Y</code> and <code>YHat</code></p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.SSRes-Tuple{Any,Any}" href="#ChemometricsTools.SSRes-Tuple{Any,Any}"><code>ChemometricsTools.SSRes</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">SSRes( y, yhat )</code></pre><p>Calculates Sum of Squared Residuals from vectors <code>Y</code> and <code>YHat</code></p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.SSTotal-Tuple{Any}" href="#ChemometricsTools.SSTotal-Tuple{Any}"><code>ChemometricsTools.SSTotal</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">SSTotal( y, yhat )</code></pre><p>Calculates Total Sum of Squared Deviations from vectors <code>Y</code> and <code>YHat</code></p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.SampleSkewness-Tuple{Any}" href="#ChemometricsTools.SampleSkewness-Tuple{Any}"><code>ChemometricsTools.SampleSkewness</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">SampleSkewness(X)</code></pre><p>returns a measure of skewness for vector <code>X</code> that is corrected for a sample of the population.</p><p>Joanes, D. N., and C. A. Gill. 1998. “Comparing Measures of Sample Skewness and Kurtosis”. The Statistician 47(1): 183–189.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.SavitzkyGolay-Tuple{Any,Any,Any,Int64}" href="#ChemometricsTools.SavitzkyGolay-Tuple{Any,Any,Any,Int64}"><code>ChemometricsTools.SavitzkyGolay</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">SavitzkyGolay(X, Delta, PolyOrder, windowsize)</code></pre><p>Performs SavitskyGolay smoothing across every row in an Array <code>X</code>. The <code>window size</code> is the size of the convolution filter, <code>PolyOrder</code> is the order of the polynomial, and <code>Delta</code> is the order of the derivative.</p><p>Savitzky, A.; Golay, M.J.E. (1964). &quot;Smoothing and Differentiation of Data by Simplified Least Squares Procedures&quot;. Analytical Chemistry. 36 (8): 1627–39. doi:10.1021/ac60214a047.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.Scale1Norm-Tuple{Any}" href="#ChemometricsTools.Scale1Norm-Tuple{Any}"><code>ChemometricsTools.Scale1Norm</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">Scale1Norm(X)</code></pre><p>Scales the columns of <code>X</code> by the 1-Norm of each row. Returns the scaled array.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.Scale2Norm-Tuple{Any}" href="#ChemometricsTools.Scale2Norm-Tuple{Any}"><code>ChemometricsTools.Scale2Norm</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">Scale2Norm(X)</code></pre><p>Scales the columns of <code>X</code> by the 2-Norm of each row. Returns the scaled array.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.ScaleInfNorm-Tuple{Any}" href="#ChemometricsTools.ScaleInfNorm-Tuple{Any}"><code>ChemometricsTools.ScaleInfNorm</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">ScaleInfNorm(X)</code></pre><p>Scales the columns of <code>X</code> by the Inf-Norm of each row. Returns the scaled array.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.ScaleMinMax-Tuple{Any}" href="#ChemometricsTools.ScaleMinMax-Tuple{Any}"><code>ChemometricsTools.ScaleMinMax</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">ScaleMinMax(X)</code></pre><p>Scales the columns of <code>X</code> by the Min and Max of each row such that no observation is greater than 1 or less than zero. Returns the scaled array.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.SecondDerivative-Tuple{Any}" href="#ChemometricsTools.SecondDerivative-Tuple{Any}"><code>ChemometricsTools.SecondDerivative</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">FirstDerivative(X)</code></pre><p>Uses the finite difference method to compute the second derivative for every row in <code>X</code>. <em>Note: This operation results in the loss of two columns.</em></p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.Shuffle!-Tuple{Any,Any}" href="#ChemometricsTools.Shuffle!-Tuple{Any,Any}"><code>ChemometricsTools.Shuffle!</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">Shuffle!( X, Y )</code></pre><p>Shuffles the rows of the <code>X</code> and <code>Y</code> data without replacement in place. In place, means that this function alters the order of the data in memory and this function does not return anything.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.Shuffle-Tuple{Any,Any}" href="#ChemometricsTools.Shuffle-Tuple{Any,Any}"><code>ChemometricsTools.Shuffle</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">Shuffle( X, Y )</code></pre><p>Shuffles the rows of the <code>X</code> and <code>Y</code> data without replacement. It returns a 2-Tuple of the shuffled set.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.SinglePointCrossOver-Tuple{BinaryLifeform,BinaryLifeform}" href="#ChemometricsTools.SinglePointCrossOver-Tuple{BinaryLifeform,BinaryLifeform}"><code>ChemometricsTools.SinglePointCrossOver</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">SinglePointCrossOver( L1::BinaryLifeform, L2::BinaryLifeform )</code></pre><p>Creates two offspring (new BinaryLifeForms) by mixing the genes from <code>L1</code> and <code>L2</code> after a random position in the vector.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.Skewness-Tuple{Any}" href="#ChemometricsTools.Skewness-Tuple{Any}"><code>ChemometricsTools.Skewness</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">Skewness(X)</code></pre><p>returns a measure of skewness for a population vector <code>X</code>.</p><p>Joanes, D. N., and C. A. Gill. 1998. “Comparing Measures of Sample Skewness and Kurtosis”. The Statistician 47(1): 183–189.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.SplitByProportion" href="#ChemometricsTools.SplitByProportion"><code>ChemometricsTools.SplitByProportion</code></a> — <span class="docstring-category">Function</span>.</div><div><div><pre><code class="language-julia">SplitByProportion(X::Array, Y::Array,Proportion::Float64 = 0.5)</code></pre><p>Splits an <code>X</code> and Associated <code>Y</code> Array along the observations dimension into a 2-Tuple of 2-Tuples based on the <code>Proportion</code>. The form of the output is the following: ( (X1, Y1), (X2, Y2) )</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.SplitByProportion" href="#ChemometricsTools.SplitByProportion"><code>ChemometricsTools.SplitByProportion</code></a> — <span class="docstring-category">Function</span>.</div><div><div><pre><code class="language-julia">SplitByProportion(X::Array, Proportion::Float64 = 0.5)</code></pre><p>Splits <code>X</code> Array along the observations dimension into a 2-Tuple based on the <code>Proportion</code>. The form of the output is the following: ( X1, X2 )</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.StandardNormalVariate-Tuple{Any}" href="#ChemometricsTools.StandardNormalVariate-Tuple{Any}"><code>ChemometricsTools.StandardNormalVariate</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">StandardNormalVariate(X)</code></pre><p>Scales the columns of <code>X</code> by the mean and standard deviation of each row. Returns the scaled array.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.StatsDictToDataFrame-Tuple{Any,Any}" href="#ChemometricsTools.StatsDictToDataFrame-Tuple{Any,Any}"><code>ChemometricsTools.StatsDictToDataFrame</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">StatsDictToDataFrame(DictOfStats, schema)</code></pre><p>Converts a dictionary of statistics which is returned from <code>MulticlassStats</code> into a labelled dataframe. This is an intermediate step for automated report generation.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.StatsFromTFPN-NTuple{4,Any}" href="#ChemometricsTools.StatsFromTFPN-NTuple{4,Any}"><code>ChemometricsTools.StatsFromTFPN</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">StatsFromTFPN(TP, TN, FP, FN)</code></pre><p>Calculates many essential classification statistics based on the numbers of True Positive(<code>TP</code>), True Negative(<code>TN</code>),  False Positive(<code>FP</code>), and False Negative(<code>FN</code>) examples.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.Threshold-Tuple{Any}" href="#ChemometricsTools.Threshold-Tuple{Any}"><code>ChemometricsTools.Threshold</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">Threshold(yhat; level = 0.5)</code></pre><p>For a binary vector <code>yhat</code> this decides if the label is a 0 or a 1 based on it&#39;s value relative to a threshold <code>level</code>.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.Update!-Tuple{RunningMean,Any}" href="#ChemometricsTools.Update!-Tuple{RunningMean,Any}"><code>ChemometricsTools.Update!</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">Update!(RM::RunningMean, x)</code></pre><p>Adds new observation(<code>x</code>) to a RunningMean object(<code>RM</code>) in place.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.Update!-Tuple{RunningVar,Any}" href="#ChemometricsTools.Update!-Tuple{RunningVar,Any}"><code>ChemometricsTools.Update!</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">Update!(RV::RunningVar, x)</code></pre><p>Adds new observation(<code>x</code>) to a RunningVar object(<code>RV</code>) and updates it in place.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.Update-Tuple{RunningMean,Any}" href="#ChemometricsTools.Update-Tuple{RunningMean,Any}"><code>ChemometricsTools.Update</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">Update!(RM::RunningMean, x)</code></pre><p>Adds new observation(<code>x</code>) to a RunningMean object(<code>RM</code>) and returns the new object.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.Variance-Tuple{ChemometricsTools.ewma}" href="#ChemometricsTools.Variance-Tuple{ChemometricsTools.ewma}"><code>ChemometricsTools.Variance</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">Variance(P::ewma)</code></pre><p>This function returns the EWMA control variance.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.Variance-Tuple{RunningVar}" href="#ChemometricsTools.Variance-Tuple{RunningVar}"><code>ChemometricsTools.Variance</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">Variance(rv::RunningVar)</code></pre><p>Returns the current variance inside of a RunningVar object.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.VenetianBlinds-Tuple{Any,Any}" href="#ChemometricsTools.VenetianBlinds-Tuple{Any,Any}"><code>ChemometricsTools.VenetianBlinds</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">VenetianBlinds(X,Y)</code></pre><p>Splits an <code>X</code> and associated <code>Y</code> Array along the observation dimension into a 2-Tuple of 2-Tuples based on the whether it is even or odd. The form of the output is the following: ( (X1,Y1), (X2, Y2) )</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.VenetianBlinds-Tuple{Any}" href="#ChemometricsTools.VenetianBlinds-Tuple{Any}"><code>ChemometricsTools.VenetianBlinds</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">VenetianBlinds(X)</code></pre><p>Splits an <code>X</code> Array along the observations dimension into a 2-Tuple of 2-Tuples based on the whether it is even or odd. The form of the output is the following: ( X1, X2 )</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.boxcar-Tuple{Any}" href="#ChemometricsTools.boxcar-Tuple{Any}"><code>ChemometricsTools.boxcar</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">boxcar(X; windowsize = 3, fn = mean)</code></pre><p>Applies a boxcar function (<code>fn</code>) to each window of size <code>windowsize</code> to every row in <code>X</code>. <em>Note: the function provided must support a dims argument/parameter.</em></p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.entropy-Tuple{Any}" href="#ChemometricsTools.entropy-Tuple{Any}"><code>ChemometricsTools.entropy</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">entropy(v)</code></pre><p>Calculates the Shannon-Entropy of a probability vector <code>v</code>. Returns a scalar. A common gain function used in tree methods.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.findpeaks-Tuple{Any}" href="#ChemometricsTools.findpeaks-Tuple{Any}"><code>ChemometricsTools.findpeaks</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">findpeaks( vY; m = 3)</code></pre><p>Finds the indices of peaks in a vector vY with a window span of <code>2m</code>. Original R function by Stas_G:(https://stats.stackexchange.com/questions/22974/how-to-find-local-peaks-valleys-in-a-series-of-data) This version is based on a C++ variant by me.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.gini-Tuple{Any}" href="#ChemometricsTools.gini-Tuple{Any}"><code>ChemometricsTools.gini</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">gini(p)</code></pre><p>Calculates the GINI coefficient of a probability vector <code>p</code>. Returns a scalar. A common gain function used in tree methods.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.offsetToZero-Tuple{Any}" href="#ChemometricsTools.offsetToZero-Tuple{Any}"><code>ChemometricsTools.offsetToZero</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">offsetToZero(X)</code></pre><p>Ensures that no observation(row) of Array <code>X</code> is less than zero, by ensuring the minimum value of each row is zero.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.plotchem-Tuple{QQ}" href="#ChemometricsTools.plotchem-Tuple{QQ}"><code>ChemometricsTools.plotchem</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">plotchem(QQ::{QQ, BlandAltman}; title )</code></pre><p>returns either a QQ Plot or a Bland-Altman plot with the defined <code>title</code></p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.rbinomial-Tuple{Any,Vararg{Any,N} where N}" href="#ChemometricsTools.rbinomial-Tuple{Any,Vararg{Any,N} where N}"><code>ChemometricsTools.rbinomial</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">rbinomial( p, size... )</code></pre><p>Makes an N-dimensional array of size(s) <code>size</code> with a probability of being a 1 over a 0 of 1 <code>p</code>.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.sigmoid-Tuple{Any}" href="#ChemometricsTools.sigmoid-Tuple{Any}"><code>ChemometricsTools.sigmoid</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">sigmoid(x)</code></pre><p>Applies the sigmoid function to a scalar value X. Returns a scalar. Can be broad-casted over an Array.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.ssd-Tuple{Any,Any}" href="#ChemometricsTools.ssd-Tuple{Any,Any}"><code>ChemometricsTools.ssd</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">ssd(p)</code></pre><p>Calculates the sum squared deviations from a decision tree split. Accepts a vector of values, and the mean of that  vector. Returns a scalar. A common gain function used in tree methods.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.DirectStandardizationXform-Tuple{Any}" href="#ChemometricsTools.DirectStandardizationXform-Tuple{Any}"><code>ChemometricsTools.DirectStandardizationXform</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">(DSX::DirectStandardizationXform)(X; Factors = length(DSX.pca.Values))</code></pre><p>Applies a the transform from a learned direct standardization object <code>DSX</code> to new data <code>X</code>.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.ELM-Tuple{Any}" href="#ChemometricsTools.ELM-Tuple{Any}"><code>ChemometricsTools.ELM</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">(M::ELM)(X)</code></pre><p>Makes an inference from <code>X</code> using a ELM object.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.KRR-Tuple{Any}" href="#ChemometricsTools.KRR-Tuple{Any}"><code>ChemometricsTools.KRR</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">(M::KRR)(X)</code></pre><p>Makes an inference from <code>X</code> using a KRR object which wraps a ClassicLeastSquares object.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.ewma-Tuple{Any}" href="#ChemometricsTools.ewma-Tuple{Any}"><code>ChemometricsTools.ewma</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">EWMA(P::ewma)(New; train = true)</code></pre><p>Provides an EWMA score for a <code>New</code> scalar value. If <code>train == true</code> the model is updated to include this new value.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.pipeline-Tuple{Any}" href="#ChemometricsTools.pipeline-Tuple{Any}"><code>ChemometricsTools.pipeline</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">(P::pipeline)(X; inverse = false)</code></pre><p>Applies the stored transformations in a pipeline object <code>P</code> to data in X. The inverse flag can allow for the transformations to be reversed provided they are invertible functions.</p></div></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="ChemometricsTools.ChangeCenter-Tuple{ChemometricsTools.ewma,Float64}" href="#ChemometricsTools.ChangeCenter-Tuple{ChemometricsTools.ewma,Float64}"><code>ChemometricsTools.ChangeCenter</code></a> — <span class="docstring-category">Method</span>.</div><div><div><pre><code class="language-julia">ChangeCenter(P::ewma, new::Float64)</code></pre><p>This is a convenience function to update the center of a <code>P</code> EWMA model, to a <code>new</code> scalar value.</p></div></div></section><footer><hr/><a class="previous" href="../GeneticAlgorithms/"><span class="direction">Previous</span><span class="title">Genetic Algorithms</span></a></footer></article></body></html>
